Tools/Schema.NET.Tool/SchemaSourceGenerator.cs

```diff
@@ -0,0 +1,198 @@
+            {
+                var schemaRepository = new SchemaRepository(new HttpClient()
+                {
+                    BaseAddress = new Uri("https://schema.org"),
+                });
+                var schemaService = new SchemaService(
+                    new IClassOverride[]
+                    {
+                        new AddQueryInputPropertyToSearchAction(),
+                        new AddTextTypeToActionTarget(),
+                        new AddNumberTypeToMediaObjectHeightAndWidth(),
+                        new RenameEventProperty(),
+                    },
+                    Array.Empty<IEnumerationOverride>(),
+                    schemaRepository);
+
+                this.SchemaObjects = await schemaService.GetObjectsAsync().ConfigureAwait(false);
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
+            }).GetAwaiter().GetResult();
+#pragma warning restore IDE0022 // Use expression body for methods
+#pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
+        }
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            foreach (var schemaObject in this.SchemaObjects)
+            {
+                var source = string.Empty;
+                if (schemaObject is GeneratorSchemaClass schemaClass)
+                {
+                    source = RenderClass(schemaClass);
+                }
+                else if (schemaObject is GeneratorSchemaEnumeration schemaEnumeration)
+                {
+                    source = RenderEnumeration(schemaEnumeration);
+                }
+
+                context.AddSource($"{schemaObject.Layer}.{schemaObject.Name}.Generated.cs", source);
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.LayoutRules", "SA1513:Closing brace should be followed by blank line", Justification = "Interpolated string")]
+        private static string RenderClass(GeneratorSchemaClass schemaClass)
+        {
+            if (schemaClass.Parents.Count > 1)
+            {
+                throw new ArgumentException(Resources.InterfaceShouldOnlyHaveOneParent);
+            }
+
+            var parentType = schemaClass.Parents.FirstOrDefault();
+
+            var interfaceImplements = string.Empty;
+            if (schemaClass.IsCombined)
+            {
+                interfaceImplements = $" : {string.Join(", ", schemaClass.CombinationOf.Select(c => $"I{c.Name}"))}";
+            }
+            else if (parentType is not null)
+            {
+                interfaceImplements = $" : I{parentType.Name}";
+            }
+
+            var classModifiers = schemaClass.IsCombined ? " abstract" : string.Empty;
+            var classImplements = parentType is not null ? $" {parentType.Name}," : string.Empty;
+            var allProperties = schemaClass.Properties.OrderBy(x => x.Order).ToArray();
+
+            return
+$@"namespace Schema.NET
```

Comment:
Will it be easy for us to spit out Pending schema classes to a Schema.NET.Pending project? I guess the generator would have to run twice, once for each project right?

Reply:
I'm not 100% sure though I was trying to keep that in mind while building this. No doubt changes to the source generator will need to happen for support to a "pending schema" project. One of the things we will be able to do is put options in the target csproj file that the source generator can read. We can use that to pass options into the `SchemaService` to choose what classes to generate.